lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -4308,6 +4308,147 @@ class ConvertAtenIndexTensorOp : public OpConversionPattern<AtenIndexTensorOp> {
+        loc, rewriter.getI64IntegerAttr(1442695040888963407));
+    // Tn = (index + Tn-1) * multiplier + incrementStep
+    auto getNextTemp = [&](OpBuilder &b, Value index, Value temp) {
+      Value castIndex =
+          b.create<arith::IndexCastOp>(loc, b.getI64Type(), index);
+      Value add = b.create<arith::AddIOp>(loc, castIndex, temp);
+      Value mult = b.create<arith::MulIOp>(loc, add, multiplier);
+      return b.create<arith::AddIOp>(loc, mult, incrementStep);
+    };
+
+    // Get initial seed, min and max used by `linalg.generic` compute payload.
+    FuncOp getter = findOrCreateDefaultGeneratorGetter(rewriter, op);
+    Value initialSeed = rewriter.create<CallOp>(loc, getter).getResult(0);
+    Value min = convertScalarToDtype(rewriter, loc, from, elemTy);
+    Value max = convertScalarToDtype(rewriter, loc, to, elemTy);
+
+    // Construct the `linalg.generic` op.
+    auto resultRank = resultType.getRank();
+    SmallVector<AffineMap, 1> indexingMaps(
+        1, rewriter.getMultiDimIdentityMap(resultRank));
+    SmallVector<StringRef> iteratorTypes(resultRank,
+                                         getParallelIteratorTypeName());
+    SmallVector<Value> sizes = getTensorSizes(rewriter, loc, self);
+    Value initTensor =
+        rewriter.create<linalg::InitTensorOp>(loc, sizes, elemTy);
+    auto res =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, resultType, /*inputs=*/ValueRange{},
+                /*outputs=*/initTensor, indexingMaps, iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value temp = initialSeed;
+                  for (int i = 0; i < resultRank; i++) {
+                    Value index = b.create<linalg::IndexOp>(loc, i);
+                    temp = getNextTemp(b, index, temp);
+                  }
+                  // Scaling = (max - min) * const(F64,  5.4210108E-20)
+                  // which is derived from rand(min,max) =
+                  // rand()/(RAND_MAX/(max-min)) where RAND_MAX = 2^64 - 1
+                  Value epsilon = b.create<arith::ConstantOp>(
+                      loc, b.getFloatAttr(min.getType(), 5.4210108E-20));
+                  Value range = b.create<arith::SubFOp>(loc, max, min);
+                  Value scale = b.create<arith::MulFOp>(loc, range, epsilon);
+
+                  // res = cast(T, cast(F64, tempN) * scaling + min)
+                  Value updateFloat =
+                      b.create<arith::UIToFPOp>(loc, elemTy, temp);
+                  Value updateScaled =
+                      b.create<arith::MulFOp>(loc, updateFloat, scale);
+                  Value res = b.create<arith::AddFOp>(loc, updateScaled, min);
+                  b.create<linalg::YieldOp>(loc, res);
+                })
+            .getResult(0);
+
+    // Set the next seed by recaculating the temp from the last iteration of
+    // `linalg.generic`.
+    Value temp = initialSeed;
```

Comment:
TorchToLinalg shouldn't need to be concerned with updating the seed. The internal implementation that a backend chooses for `torch_c.get_next_seed` is responsible for that.